perun-core/src/main/java/cz/metacentrum/perun/core/blImpl/AuthzResolverBlImpl.java

```diff
@@ -1843,6 +1843,134 @@ public static PerunBl setPerunBl(PerunBl perunBl) {
 		return perunBl;
 	}
 
+	/**
+	 * Get all Vos where the given user has set one of the given roles
+	 * or the given user is a member of an authorized group with such roles.
+	 *
+	 * @param sess Perun session
+	 * @param user for who Vos are retrieved
+	 * @param roles for which Vos are retrieved
+	 * @return List of Vos
+	 */
+	public static List<Vo> getVosWhereUserIsInRoles(PerunSession sess, User user, List<String> roles) {
+		for (String role: roles) {
+			if (!roleExists(role)) {
+				throw new InternalErrorException("Role: "+ role +" does not exists.");
+			}
+		}
+
+		return authzResolverImpl.getVosWhereUserIsInRoles(user, roles);
+	}
+
+	/**
+	 * Get all Facilities where the given user has set one of the given roles
+	 * or the given user is a member of an authorized group with such roles.
+	 *
+	 * @param sess Perun session
+	 * @param user for who Facilities are retrieved
+	 * @param roles for which Facilities are retrieved
+	 * @return List of Facilities
+	 */
+	public static List<Facility> getFacilitiesWhereUserIsInRoles(PerunSession sess, User user, List<String> roles) {
+		for (String role: roles) {
+			if (!roleExists(role)) {
+				throw new InternalErrorException("Role: "+ role +" does not exists.");
+			}
+		}
+
+		return authzResolverImpl.getFacilitiesWhereUserIsInRoles(user, roles);
+	}
+
+	/**
+	 * Get all Resources where the given user has set one of the given roles
+	 * or the given user is a member of an authorized group with such roles.
+	 *
+	 * @param sess Perun session
+	 * @param user for who Resources are retrieved
+	 * @param roles for which Resources are retrieved
+	 * @return List of Resources
+	 */
+	public static List<Resource> getResourcesWhereUserIsInRoles(PerunSession sess, User user, List<String> roles) {
+		for (String role: roles) {
+			if (!roleExists(role)) {
+				throw new InternalErrorException("Role: "+ role +" does not exists.");
+			}
+		}
+
+		return authzResolverImpl.getResourcesWhereUserIsInRoles(user, roles);
+	}
+
+	/**
+	 * Get all Groups where the given user has set one of the given roles
+	 * or the given user is a member of an authorized group with such roles.
+	 *
+	 * @param sess Perun session
+	 * @param user for who Groups are retrieved
+	 * @param roles for which Groups are retrieved
+	 * @return List of Groups
+	 */
+	public static List<Group> getGroupsWhereUserIsInRoles(PerunSession sess, User user, List<String> roles) {
+		for (String role: roles) {
+			if (!roleExists(role)) {
+				throw new InternalErrorException("Role: "+ role +" does not exists.");
+			}
+		}
+
+		Set<Group> result = new HashSet<>();
+		List<Group> parentGroups = authzResolverImpl.getGroupsWhereUserIsInRoles(user, roles);
+
+		for (Group parentGroup: parentGroups) {
+			result.add(parentGroup);
+			result.addAll(perunBl.getGroupsManagerBl().getAllSubGroups(sess, parentGroup));
```

Comment:
I am not really sure if we want this behavior. It might be a bit confusing. Image a rich tree group structure and some user who has been given a GROUPADMIN of the root group. In GUI, he will be shown, that he is an admin of something like 1000 groups and he will not be able to distinguish between the root group and subgroups.

Reply:
You are right. Removed.